### PR TITLE
implement per-route override for rbac stat prefixes

### PR DESCRIPTION
**Pull request by antoniovleonti**
Commit Message: implement per-route override for rbac stat prefixes
Additional Description: currently, if you only have stat prefixes configured in per-route config and not the base config, the stat prefix will be empty. this PR addresses that.
Risk Level: low
Testing: unit tested
Docs Changes: n/a
Release Notes: changelog updated

### CODE REVIEW

**Review by yanavlasov — APPROVED** [review]

**Comment by yanavlasov** [comment]
Will wait for @yangminzhu review before merging.

**Comment by rshriram** [comment]
@yangminzhu gentle ping

File: source/extensions/filters/http/rbac/rbac_filter.cc
Comment (yangminzhu) [inline source/extensions/filters/http/rbac/rbac_filter.cc:152]: per_route_config.rbac().rules_stat_prefix() will be empty if not specified right? just double checking to make sure this won't change existing behavior.

File: source/extensions/filters/http/rbac/rbac_filter.cc
Comment (yangminzhu) [inline source/extensions/filters/http/rbac/rbac_filter.cc:74]: the 4 functions all have the same pattern, wondering if we could do something (like templating) to avoid duplicating the code?

File: source/extensions/filters/http/rbac/rbac_filter.cc
Comment (antoniovleonti) [reply source/extensions/filters/http/rbac/rbac_filter.cc:127]: Good catch, I fixed this to be backward compatible and added a test.

**Comment by yanavlasov** [comment]
@antoniovleonti it looks like main merge is needed to address coverage failure.

**Review by yanavlasov — APPROVED** [review]

**Review by yangminzhu — APPROVED** [review]
thanks!

**Pull request merged**
